File: IOS/IOS 8.2/FishBowlLibrary-NewCRMAPIIntegration/Bistro/JambaApp/JMBA_JambaJuice_Latest/JambaJuice/BillingAccount.swift
//
//  BillingAccount.swift
//  JambaJuice
//
//  Created by Taha Samad on 6/23/15.
//  Copyright (c) 2015 Jamba Juice. All rights reserved.
//

import Foundation
import OloSDK

struct BillingAccount {
    
    var accountId: Int64
    var accountType: String
    var cardType: String
    var cardSuffix: String
    var desc: String
    var expiration: String
    
    init(oloBillingAccount: OloBillingAccount) {
        accountId   = oloBillingAccount.accountId
        accountType = oloBillingAccount.accountType
        cardType    = oloBillingAccount.cardType
        cardSuffix  = oloBillingAccount.cardSuffix
        desc        = oloBillingAccount.desc
        expiration  = oloBillingAccount.expiration
    }
}

File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.1/JMBA_JambaJuice_Android/app/src/main/java/com/olo/jambajuice/BusinessLogic/Models/BillingAccount.java
package com.olo.jambajuice.BusinessLogic.Models;

import com.olo.jambajuice.Utils.StringUtilities;
import com.wearehathway.apps.olo.Models.OloBillingAccount;

/**
 * Created by Nauman Afzaal on 18/06/15.
 */
public class BillingAccount
{

    private int accountid;
    private String accounttype;
    private String cardtype;
    private String cardSuffix;
    private String description;
    private String expiration;

    public BillingAccount(OloBillingAccount oloBillingAccount)
    {
        accountid = oloBillingAccount.getAccountid();
        cardtype = oloBillingAccount.getCardtype();
        cardSuffix = oloBillingAccount.getCardsuffix();
        accounttype = oloBillingAccount.getAccounttype();
        description = oloBillingAccount.getDescription();
        expiration = oloBillingAccount.getExpiration();
    }

    public void setCardtype(String cardtype) {
        this.cardtype = cardtype;
    }

    public int getAccountid()
    {
        return accountid;
    }

    public String getCardtype()
    {
        String cardtypee = "";
        if(StringUtilities.isValidString(cardtype)){
            cardtypee = cardtype;
        }else if(accounttype.equalsIgnoreCase("payinstore")){
            cardtypee = "Pay Restaurant Directly";
        }
        return cardtypee;
    }

    public String getCardSuffix()
    {
        String cardSuffixx = "";
        if(StringUtilities.isValidString(cardSuffix)) {
            cardSuffixx =  cardSuffix;
        }else if(accounttype.equalsIgnoreCase("payinstore")){
            cardSuffixx = "payinstore";

        }
        return cardSuffixx;
    }

    public String getAccounttype() {
        return accounttype;
    }

    public void setAccounttype(String accounttype) {
        this.accounttype = accounttype;
    }
}
